src/main/java/slimeknights/tconstruct/smeltery/block/entity/module/MeltingModuleInventory.java

```diff
@@ -127,6 +126,25 @@ public int getRequiredTemp(int slot) {
     return hasModule(slot) ? modules[slot].getRequiredTemp() : 0;
   }
 
+  /**
+   * Finds a melting recipe
+   * @param module Melting Module asking for a recipe
+   * @param slotRecipe Last used Recipe of that slot
+   * @return Melting recipe found, or null if no match
+   */
+  @Nullable
+  public IMeltingRecipe findRecipe(MeltingModule module, @Nullable IMeltingRecipe slotRecipe) {
+    Level world = parent.getLevel();
+    if (world == null) {
+      return null;
+    }
+    if (lastInventoryRecipe != null && slotRecipe != lastInventoryRecipe && lastInventoryRecipe.matches(module, world)) {
+        return lastInventoryRecipe;
+    }
+    Optional<IMeltingRecipe> newRecipe = world.getRecipeManager().getRecipeFor(TinkerRecipeTypes.MELTING.get(), module, world);
```

Comment:
the whole point of having a cache is to save the recipe in the cache when found, you should do so.